src/controllers/roomController.js

```diff
@@ -0,0 +1,97 @@
+import roomService from '../services/room.service';
+import ErrorResponse from '../utils/errorResponse';
+
+class RoomController {
+  static createRoom = async (req, res) => {
+    try {
+      console.log(typeof req.body);
+      const room = await roomService.createRoom(req.body);
+      return res.status(201).json({
+        status: 201,
+        message: 'room added successfully',
+        payload: room,
+      });
+    } catch (error) {
+      ErrorResponse.internalServerError(res, error.message);
+    }
+  };
+
+  static getAllRoomsOfAccommodation = async (req, res) => {
+    try {
+      const foundRooms = await roomService.getAllRoomsOfAccommodation(
+        req.params.accommodationId
+      );
+      return res.status(200).json({
+        status: '200',
+        message: 'All rooms in given accommodation',
+        payload: foundRooms,
+      });
+    } catch (error) {
+      ErrorResponse.internalServerError(res, error.message);
+    }
+  };
+
+  static getSingleRoom = async (req, res) => {
+    try {
+      const foundRoom = await roomService.getSingleRoom(req.params.roomId);
+      if (foundRoom) {
+        return res.status(200).json({
+          status: '200',
+          message: 'Room found',
+          payload: foundRoom,
+        });
+      }
+      return res.status(404).json({
+        status: '404',
+        message: 'Room not found',
+      });
+    } catch (error) {
+      ErrorResponse.internalServerError(res, error.message);
+    }
+  };
+
+  static updateRoom = async (req, res) => {
+    try {
+      await roomService.updateRoom(req.params.roomId, req.body);
+
+      // if (updatedRoom[0] === 0) {
+      //   return res.status(404).json({
```

Comment:
why did you comment this out?
if you no longer need it remove it

Reply:
if you are deleting it, then i think you can work on test coz i can see tests are failing then you PRget merged.

Reply:
Those lines of codes were returning an error upon not finding any room while an empty array is indeed a valid response.